docs/content/docs/reference/react/protecting-routes.mdx

```diff
@@ -4,49 +4,67 @@ title: 'Protecting routes'
 
 ## React Router
 
-You can protect routes by creating an `AuthGate` component when using `@nhost/react` with [React Router](https://reactrouter.com/web/guides/quick-start).
+You can protect routes by creating a wrapper component, `ProtectedRoute`, to implement the authentication logic using `@nhost/react`.
+
+> This example uses the latest version of [React Router (v6)](https://reactrouter.com/docs/en/v6).
 
 ```jsx
-import { useAuthenticationStatus } from '@nhost/react'
-import { Redirect } from 'react-router-dom'
+// src/components/ProtectedRoute.js
+
+import { useAuthenticationStatus } from '@nhost/react';
+import { Navigate, useLocation } from 'react-router';
 
-export function AuthGate({ children }) {
-  const { isLoading, isAuthenticated } = useAuthenticationStatus()
+const ProtectedRoute = ({ children }) => {
+  const { isAuthenticated, isLoading } = useAuthenticationStatus();
+  const location = useLocation();
 
   if (isLoading) {
-    return <div>Loading...</div>
+    return (
+      <div>Loading...</div>
+    );
   }
 
   if (!isAuthenticated) {
-    return <Redirect to="/login" />
+    return <Navigate to="/login" state={{ from: location }} replace />;
   }
 
-  return children
-}
+  return children;
```

Comment:
Could you please change this to `<Outlet />`? React Router renders children of layouts via outlets.

Reply:
You're right. I've mixed things up since I first wrote the example with an extra `Layout` component that rendered the `Outlet`. But for simplifying the example, I removed it and only used the `ProtectedRoute`.

I'm pushing the changes.